src/victory-legend/victory-legend.js

```diff
@@ -0,0 +1,252 @@
+import React, { PropTypes } from "react";
+import { PropTypes as CustomPropTypes, TextSize, Helpers } from "../victory-util/index";
+import { merge, isEmpty, defaults, sumBy, maxBy } from "lodash";
+import VictoryLabel from "../victory-label/victory-label";
+import VictoryContainer from "../victory-container/victory-container";
+import Point from "../victory-primitives/point";
+
+const defaultStyles = {
+  symbol: {
+    fill: "black",
+    type: "circle"
+  },
+  label: {
+    fontSize: 14,
+    fontFamily: "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
+    color: "#252525",
+    backgroundColor: "#d9d9d9",
+    stroke: "transparent"
+  }
+};
+const defaultLegendData = [
+  { name: "Series 1" },
+  { name: "Series 2" }
+];
+
+export default class VictoryLegend extends React.Component {
+  static displayName = "VictoryLegend";
+  static role = "legend";
+
+  static propTypes = {
+    x: PropTypes.number.isRequired,
+    y: PropTypes.number.isRequired,
+    width: PropTypes.oneOfType([
+      CustomPropTypes.nonNegative,
+      PropTypes.func
+    ]),
+    height: PropTypes.oneOfType([
+      CustomPropTypes.nonNegative,
+      PropTypes.func
+    ]),
+    padding: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.shape({
+        top: PropTypes.number,
+        bottom: PropTypes.number,
+        left: PropTypes.number,
+        right: PropTypes.number
+      })
+    ]),
+    orientation: PropTypes.oneOf(["horizontal", "vertical"]),
+    data: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        label: PropTypes.object,
+        symbol: PropTypes.object
+      })
+    ),
+    containerComponent: PropTypes.element,
+    dataComponent: PropTypes.element,
+    groupComponent: PropTypes.element,
+    labelComponent: PropTypes.element,
+    symbolSpacer: PropTypes.number,
+    gutter: PropTypes.number,
+    standalone: PropTypes.bool,
+    style: PropTypes.shape({
+      symbol: PropTypes.object,
+      label: PropTypes.object
+    })
+  };
+
+  static defaultProps = {
+    x: 0,
+    y: 0,
+    padding: 0,
+    symbolSpacer: 8,
+    gutter: 10,
+    orientation: "vertical",
+    dataComponent: <Point/>,
+    labelComponent: <VictoryLabel/>,
+    containerComponent: <VictoryContainer/>,
+    groupComponent: <g/>,
+    standalone: true,
+    style: {}
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = this.getLegendState(props);
```

Comment:
Other components calculate values in render and then pass some kind of `calculatedProps` object down to the methods that need them.